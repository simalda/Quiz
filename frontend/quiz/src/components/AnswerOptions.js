import React from "react";

import "./QuizStep";

function AnswerOptions({ title, highlight, onClick, index }) {
  let className =
    "top10 answers button  " + (!!highlight ? highlight : "notSelected");
  let id = "answer" + index;

  return (
    <div id={id} className={className} onClick={() => onClick(title)}>
      <h4>{title}</h4>
    </div>
  );
}

export default AnswerOptions;
