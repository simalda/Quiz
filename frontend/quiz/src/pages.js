const pages = {
    Login: "Login",//-1
    NumberofQuestions:"NumberofQuestions",//3
    Main: "Main",//1
    QuizStep: "QuizStep",//0
    Results:"Results",//2
    Statistics: "Statistics",//5
    AddQuestion:"AddQuestion",
    AboutUs: "AboutUs"
  };
  
  export default pages