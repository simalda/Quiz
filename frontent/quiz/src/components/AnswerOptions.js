import React, { Component } from "react";
import ReactDOM from "react-dom";
import "../index.css";

function AnswerOptions({ title, highlight, onClick }) {
  var clasname = "answer " + (!!highlight ? highlight : "");

  return (
    <div id="answer" className={clasname} onClick={() => onClick(title)}>
      <h4>{title}</h4>
    </div>
  );
}

export default AnswerOptions;
