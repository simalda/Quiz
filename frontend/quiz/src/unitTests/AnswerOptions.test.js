import React from "react";
import ReactDOM from "react-dom";

import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import AnswerOptions from "../components/AnswerOptions";
import pages from "../JS/pages";

import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
const mockStore = configureMockStore();
const store = mockStore({});
Enzyme.configure({ adapter: new Adapter() });

const state = {
  quiz: [
    {
      question: "a",
      answer: ["c", "d", "s"],
      correctAnswer: "cor"
    },
    {
      question: "ab",
      answer: ["c", "d", "s"],
      CorrectAnswer: "cor2"
    }
  ],
  highlight: "none",
  page: pages.Login,
  answerOptions: []
};

describe.skip("Answer options", () => {
  let wrapper;
  let mockSubmit;
  beforeEach(() => {
    mockSubmit = jest.fn();
    wrapper = shallow(
      <Provider store={store}>
        <AnswerOptions submit={mockSubmit} />
      </Provider>
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Page is correct", () => {
    expect(state.page).toBe(pages.Login);
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<AnswerOptions {...state} />, div);
  });
});
