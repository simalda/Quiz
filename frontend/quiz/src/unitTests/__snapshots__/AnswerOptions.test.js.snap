// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer options should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`Login should match the snapshot 1`] = `undefined`;
